Fixed broken install process on MacOS/Linux

raghavmecheri opened the pull request:
1. The patch application fails on macOS and Linux both, and the process crashes out with a deceptive error message. To be honest, I'm not entirely sure what the purpose of the patch being applied is, but it looks like a lot of those changes have been merged into the vision repository. Maybe we could do away with this process?
2. Strangely enough, C++ 14 doesn't seem to pick up `putenv` -- however, `setenv` seems to work just fine.
3. The `bin/` directory for the vision build does not seem to be generated. However, `lib/` does exist, which is where the `lylib` files live. So I copy over from that directory instead.

I'm not sure if this is related or unrelated, but with this process, I see `TypeError: c.toObject is not a function` while trying to execute `node tests/runTorch.js`

raghavmecheri committed:
Removed patch application and remove patches folder

raghavmecheri committed:
Replaced putenv with setenv

raghavmecheri committed:
Fixed issue with bin/ directory not being found while building vision

raghavmecheri commented:
@arition I've spent some time on this, but I'm not sure where the seg fault's coming from. The error looks like this:
```
TypeError: c.toObject is not a function
    at Object.<anonymous> (/Users/raghav/Projects/pytorch-js/torch-js/tests/runTorch.js:19:15)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:829:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
Segmentation fault: 11
```
Would appreciate it if you have any ideas -- if we're able to fix this and you think the rest looks good, I'd also be willing to work on a unit test suite for this project

arition committed:
Add condition switch for Win32 and MacOS/Linux

raghavmecheri committed:
Added patchfile back after control switch based on OS

raghavmecheri committed:
Merge branch 'master' of https://github.com/raghavmecheri/torch-js

arition committed:
Conditional compiling for putenv/setenv

arition committed:
fix x64 compiling on MacOS/Linux

arition commented:
@raghavmecheri Thanks for your contribution! All of three issues you mentioned do exists on MacOS/Linux, but these are essential parts on Windows. I changed your code and added conditional switches for them. If that looks good to you, I can merge the pr.

arition commented:
For the seg fault problem
https://github.com/arition/torch-js/blob/67136eef1b4dcb211200a86979dcf599bc89b542/tests/runTorch.js#L18
I changed the forward function to return a promise, but I forgot to change test file. Adding ```await``` ketword before the forward function should fix the problem.

raghavmecheri commented:
> For the seg fault problem
> https://github.com/arition/torch-js/blob/67136eef1b4dcb211200a86979dcf599bc89b542/tests/runTorch.js#L18
> 
> 
> I changed the forward function to return a promise, but I forgot to change test file. Adding `await` keyword before the forward function should fix the problem.

Got it -- let me fix that, ensure that this builds, and then push a final commit? We should be good to merge after that

raghavmecheri commented:
The `"${CMAKE_SIZEOF_VOID_P}" STREQUAL "4"` condition doesn't work as expected in my case -- I'm on a 64-bit machine, but a lib64 build doesn't exist. OSX lib files can use multiple architectures, as detailed [here](https://github.com/italiangrid/voms/issues/3) -- we can probably put a condition around this, will push a change for the same.

raghavmecheri committed:
Fixed MacOS 64 bit machine issue
Added a secondary condition when we're moving the files around post
compile. Everything builds on MacOS now.

raghavmecheri commented:
@arition We should be good to go now.
I also patched `tests/runTorch.js` -- admittedly this should have been a separate commit, sorry about that haha

arition commented on package.json:
```diff
@@ -10,10 +10,11 @@
     "compile": "cmake-js compile",
-    "rebuild": "cmake-js rebuild"
+    "rebuild": "cmake-js rebuild",
+    "clean": "rm -rf libtorch-cpu.zip"
```
I don't think we need the clean scripts here. For other platforms, the downloaded zip may have a different name. The cmake-js will also automatically redownload the zip if the build process failed.

raghavmecheri replied:
Fair enough. I had it on there because it wasn't auto-redownloading on MacOS - that's something that we should probably look into individually though.

arition committed:
format file

raghavmecheri committed:
removed clean command from package.json

raghavmecheri committed:
Merge branch 'master' of https://github.com/raghavmecheri/torch-js

arition commented:
LGTM
